### PR TITLE
add subsidence to gcm-driven setup

### LINKED ISSUE #2917: Add subsidence tendency in GCM driven SCM

### PR DESCRIPTION
**Pull request by haakon-e**
Part of #2901 

closes #2917

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/initial_conditions/initial_conditions.jl
Comment (szy21) [inline src/initial_conditions/initial_conditions.jl:1]: We can remove the changes in this file for this PR.

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 1]
```diff
@@ -113,7 +114,24 @@ function external_forcing_tendency!(Yₜ, Y, p, t, colidx, ::GCMForcing)
                 Lv_0 - R_v * T_0
             ) * ᶜdqtdt_sum[colidx]
         )
+    # total humidity
     @. Yₜ.c.ρq_tot[colidx] += Y.c.ρ[colidx] * ᶜdqtdt_sum[colidx]
 
+    ## subsidence -->
+    toa(f) = Spaces.level(f, Spaces.nlevels(axes(f)))  # get value at top-of-atmosphere
```
Comment (szy21) [inline src/prognostic_equations/forcing/external_forcing.jl:121]:
```suggestion
    tom(f) = Spaces.level(f, Spaces.nlevels(axes(f)))  # get value at top of the model
```

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 2]
```diff
@@ -113,7 +114,24 @@ function external_forcing_tendency!(Yₜ, Y, p, t, colidx, ::GCMForcing)
...
+    wvec = Geometry.WVector
+    ∇c = Operators.DivergenceF2C()
+    ∇(x) = begin
+        RB = Operators.RightBiasedC2F(; top = Operators.SetValue(toa(x)))
+        ∇c(wvec(RB(x)))
```
Comment (szy21) [inline src/prognostic_equations/forcing/external_forcing.jl:126]:
```suggestion
        ᶜdivᵥ(wvec(RB(x)))
```
(`ᶜdivᵥ` is already defined in `src/utils/abbreviations.jl`, we can clean up subsidence for other case later.)

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 3]
```diff
@@ -113,7 +114,24 @@ function external_forcing_tendency!(Yₜ, Y, p, t, colidx, ::GCMForcing)
                 Lv_0 - R_v * T_0
             ) * ᶜdqtdt_sum[colidx]
         )
+    # total humidity
     @. Yₜ.c.ρq_tot[colidx] += Y.c.ρ[colidx] * ᶜdqtdt_sum[colidx]
 
+    ## subsidence -->
+    toa(f) = Spaces.level(f, Spaces.nlevels(axes(f)))  # get value at top-of-atmosphere
+    wvec = Geometry.WVector
+    ∇c = Operators.DivergenceF2C()
```
Comment (szy21) [inline src/prognostic_equations/forcing/external_forcing.jl:123]:
```suggestion
```

File: src/surface_conditions/surface_setups.jl
Comment (szy21) [inline src/surface_conditions/surface_setups.jl:1]: We can remove the change in this file.

File: src/surface_conditions/surface_state.jl
Comment (szy21) [inline src/surface_conditions/surface_state.jl:1]: We can remove the changes in this file.

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 4]
```diff
@@ -113,7 +114,24 @@ function external_forcing_tendency!(Yₜ, Y, p, t, colidx, ::GCMForcing)
...
+    ∇(x) = begin
+        RB = Operators.RightBiasedC2F(; top = Operators.SetValue(toa(x)))
+        ∇c(wvec(RB(x)))
+    end
+    @. Yₜ.c.ρe_tot[colidx] -=
+        Y.c.ρ[colidx] * ᶜls_subsidence[colidx] * ∇(ᶜh_tot[colidx])  # ρ⋅w⋅∇h_tot
+    @. Yₜ.c.ρq_tot[colidx] -=
+        Y.c.ρ[colidx] *
+        ᶜls_subsidence[colidx] *
+        ∇(Y.c.ρq_tot[colidx] / Y.c.ρ[colidx])  # ρ⋅w⋅∇q_tot
```
Comment (szy21) [inline src/prognostic_equations/forcing/external_forcing.jl:133]:
```suggestion
        ∇(ᶜspecific.q_tot[colidx])  # ρ⋅w⋅∇q_tot
```

File: src/prognostic_equations/forcing/external_forcing.jl [hunk 5]
```diff
@@ -113,7 +114,24 @@ function external_forcing_tendency!(Yₜ, Y, p, t, colidx, ::GCMForcing)
                 Lv_0 - R_v * T_0
             ) * ᶜdqtdt_sum[colidx]
         )
+    # total humidity
```
Comment (szy21) [inline src/prognostic_equations/forcing/external_forcing.jl:117]:
```suggestion
    # total specific humidity
```

**Review by szy21 — APPROVED** [review]
Thanks! I can look at why the simulation fails tomorrow.

**Review by szy21 — APPROVED** [review]
Thank you!

**Review by costachris — APPROVED** [review]

**Pull request merged**